app/views/home.py

```diff
@@ -72,13 +74,49 @@ def send_email():
 
 @bp.route("/appointment")
 def appointment():
-    booked_appointments = appintments_model.get_booked_slots()
+    booked_appointments = appointment_model.get_booked_slots()
     timeslots = [a["timeslot"].date() for a in booked_appointments]
-    print(timeslots)
-    data = {"appointments": timeslots}
+    locked_slots = {}
+    # locked_slots: Is for dynamic lock days on the calendar in the front-end
+    for a in booked_appointments:
+        service = services_model.get(a["service_id"])
+        if service["max_slot"]:
+            if service["id"] not in locked_slots:
+                locked_slots[service["id"]] = []
+            locked_slots[service["id"]].append(
+                str(a["timeslot"].date())
+            )
+    shops = shops_model.all() or []
+    services = services_model.all() or []
+    data = {
+        "shops": shops,
+        "services": services,
+        "appointments": timeslots,
+        "locked_slots": locked_slots
+    }
     return render_template("appointment.html", config=config, data=data)
 
 
+@bp.route("/appointment", methods=["POST"])
+def save_appointment():
+    appointment_model.insert(
```

Comment:
found one issue here, if the insertion failed an email would still be sent. I think we should do some error handling here